runtime/compiler/control/CompileBeforeCheckpoint.cpp

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "j9cfg.h"
+
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+
+#include "j9protos.h"
+#include "j9nonbuilder.h"
+#include "control/CompilationController.hpp"
+#include "control/CompilationRuntime.hpp"
+#include "control/CompileBeforeCheckpoint.hpp"
+#include "control/OptimizationPlan.hpp"
+#include "env/Region.hpp"
+#include "env/VerboseLog.hpp"
+#include "env/VMJ9.h"
+
+TR::CompileBeforeCheckpoint::CompileBeforeCheckpoint(TR::Region &region, J9VMThread *vmThread, TR_J9VMBase *fej9, TR::CompilationInfo *compInfo) :
+   _region(region),
+   _vmThread(vmThread),
+   _fej9(fej9),
+   _compInfo(compInfo),
+   _methodList(_region),
+   _seenMethodsSet(std::less<TR_OpaqueMethodBlock*>(), _region)
+   {}
+
+void
+TR::CompileBeforeCheckpoint::addMethodToList(TR_OpaqueMethodBlock *method)
+   {
+   if (method && _seenMethodsSet.find(method) == _seenMethodsSet.end())
+      {
+      _methodList.push_back(method);
```

Comment:
Do we need both the list and the set of methods? We can iterate over the set, so why isn't the set enough?

Reply:
yeah that's a fair point, I think I had started off with the list and then realized I needed a way to prevent duplicates, so introduced the set. I'll remove the list.

Reply:
Updated in https://github.com/eclipse-openj9/openj9/compare/d5850691d5686f23e30b6437904cf504f3a2a87e..fac845ac4f51b4ee18576fb1eb00fb9e002af68d